src/ImageSharp.Drawing/Pens/Pen{TColor}.cs

```diff
@@ -208,7 +207,7 @@ public override ColoredPointInfo<TColor> GetColor(PointInfo info)
                 // we can treat the DistanceAlongPath and DistanceFromPath as x,y coords for the pattern
                 // we need to calcualte the distance from the outside edge of the pattern
                 // and set them on the ColoredPointInfo<TColor> along with the color.
-                infoResult.Color = this.brush.GetColor(info.SearchPoint);
+                infoResult.Color = this.brush[x, y];
```

Comment:
`ref` returns won't help too much if we use them with abstract methods here. We really shouldn't have abstract methods operating on per-pixel granularity. It's a massive cache-efficiency killer on a hot (burning!) execution path.

I'm thinking on sane ways to enable this optimization technique in ImageSharp. If it gets it's way into the library, hope we will be able to bring it into SixLabors.Shapes too, so we can boost ImageSharp drawing perf.